src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -233,22 +233,24 @@ public async Task<HttpResponseMessage> SendAsync(CancellationToken cancellationT
 
                 return response;
             }
-            catch (QuicStreamAbortedException ex) when (ex.ErrorCode == (long)Http3ErrorCode.VersionFallback)
```

Comment:
I think we had this same discussion in https://github.com/dotnet/runtime/pull/47197#issuecomment-763607980. The relevant reply copied here:

> Undoing use of exception filters sounds like a step backward to me.

cc @jkotas @stephentoub @marek-safar

Reply:
Yeah, no doubt.  This is intended as an intermediate fix.

Reply:
Why not revert the change that broke it instead?

Reply:
Looking at the IL posted in https://github.com/dotnet/runtime/issues/57010, it looks like illink decided that QuicStreamAbortedException can never be thrown, tried to eliminate this code based on that knowledge, and messed it up.

Adding this somewhere in the method (at the beginning would be a good spot) would likely work around because then illink won't try to touch the filter. It's not smart enough to see this is unreached:

```csharp
#if MONO
if (string.Empty.Length > 0)
    throw new QuicStreamAbortedException("", 0);
#endif
```

Reply:
@MichalStrehovsky Your suggestion didn't seem to work for me.  Is there something else I can try?

Reply:
If you have the artifacts around, could you attach the System.Net.Http.dll after my workaround? Ideally both the version before/after trimming?

It's possible there are more instances of this in the method body - I want to see if there's any more diffs in the method body after trimming - that would point to all the possible sources of trouble.

Reply:
> #if MONO

this is System.Net.Http - there is no `MONO` define.

Reply:
> this is System.Net.Http - there is no MONO define.

Sorry for the confusion :(. Glad the workaround works.